tests: net: socket: register: net.socket.register.tls fails at nrf5340/ns and nrf9160/ns

**Describe the bug**
Tests 
- tests/net/socket/register/net.socket.register.tls

Observed for:
- `nrf5340dk/nrf5340/cpuapp/ns`
- `nrf9160dk@0.14.0/nrf9160/ns`

**To Reproduce**
Steps to reproduce the behavior:
1. have nrf5340dk/nrf5340/cpuapp/ns connected
2. go to your zephyr dir
3. call `./scripts/twister -T tests/net/socket/register/ -p nrf5340dk/nrf5340/cpuapp/ns --device-testing --device-serial /dev/ttyACM0 -v --inline-logs --west-flash="--erase"`
4. See console output with error

**Expected behavior**
Valid console output

**Impact**
Not clear

**Logs and console output**
```
twister-out/nrf9160dk@0.14.0_nrf9160_ns/zephyr/tests/net/socket/register/net.socket.register.tls/register/tfm/api_ns/interface/include/mbedtls/check_config.h:350:2: error: #error "MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED defined, but not all prerequisites"
  350 | #error "MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED defined, but not all prerequisites"
      |  ^~~~~
```

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04.1 LTS
 - Toolchain Zephyr SDK 0.17.0
 - Commit SHA or Version used: `4a15930b0e5b441c3700dbce5d198ace7c80cd4b`

--- a/modules/mbedtls/Kconfig.mbedtls
+++ b/modules/mbedtls/Kconfig.mbedtls
@@ -60,6 +60,23 @@ config MBEDTLS_GENPRIME_ENABLED
 
 endif # MBEDTLS_RSA_C
 
+config MBEDTLS_RSA_ENABLE_LEGACY_APIS
+	bool
+	imply MBEDTLS_RSA_C
+	imply MBEDTLS_PKCS1_V15
+	default y if MBEDTLS_CIPHERSUITE_TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256 || \
+		     MBEDTLS_CIPHERSUITE_TLS_RSA_WITH_AES_256_CBC_SHA256
+	help
+	  Promptless Kconfig to automatically enable legacy RSA crypto support
+	  when RSA based ciphersuites are enabled.
+	  This is required because TLS/X.509 code still relies on these legacy
+	  symbols to guard parts of the code.
+	  In theory this fix is only necessary when TF-M is enabled, because
+	  otherwise when Mbed TLS uses its own implementation of PSA Crypto
+	  it automatically does this internally. On the other hand it doesn't hurt to
+	  always enable it.
+	  This fix is very likely to be removed with Mbed TLS 4.x.
+
 config MBEDTLS_KEY_EXCHANGE_ALL_ENABLED
 	bool "All available ciphersuite modes"
 	select MBEDTLS_MD_C

